incubator/opentelemetry-sampler-aws-xray/src/sampling-rule-applier.ts

```diff
@@ -18,18 +18,175 @@
+  ATTR_AWS_ECS_CLUSTER_ARN,
+  ATTR_AWS_ECS_CONTAINER_ARN,
+  ATTR_AWS_EKS_CLUSTER_ARN,
+  CLOUD_PLATFORM_VALUE_AWS_LAMBDA,
+  ATTR_AWS_LAMBDA_INVOKED_ARN,
+  ATTR_CLOUD_RESOURCE_ID,
+} from './semconv';
 import { ISamplingRule, SamplingTargetDocument } from './types';
 import { SamplingRule } from './sampling-rule';
 import { Statistics } from './statistics';
+import { CLOUD_PLATFORM_MAPPING, attributeMatch, wildcardMatch } from './utils';
 
 export class SamplingRuleApplier {
   public samplingRule: SamplingRule;
+  private fixedRateSampler: TraceIdRatioBasedSampler;
+  private statistics: Statistics;
 
   constructor(
     samplingRule: ISamplingRule,
     statistics: Statistics = new Statistics(),
     target?: SamplingTargetDocument
   ) {
     this.samplingRule = new SamplingRule(samplingRule);
+
+    this.fixedRateSampler = new TraceIdRatioBasedSampler(
+      this.samplingRule.FixedRate
+    );
+    // TODO: Add Reservoir Sampler (Rate Limiting Sampler)
+
+    this.statistics = statistics;
+    this.statistics.resetStatistics();
+
+    // TODO: Update Sampling Targets using provided `target` parameter
+  }
+
+  public matches(attributes: Attributes, resource: Resource): boolean {
+    let httpTarget: AttributeValue | undefined = undefined;
+    let httpUrl: AttributeValue | undefined = undefined;
+    let httpMethod: AttributeValue | undefined = undefined;
+    let httpHost: AttributeValue | undefined = undefined;
+    let serviceName: AttributeValue | undefined = undefined;
+
+    if (attributes) {
+      httpTarget = attributes[ATTR_HTTP_TARGET] ?? attributes[ATTR_URL_PATH];
+      httpUrl = attributes[ATTR_HTTP_URL] ?? attributes[ATTR_URL_FULL];
+      httpMethod =
+        attributes[ATTR_HTTP_METHOD] ?? attributes[ATTR_HTTP_REQUEST_METHOD];
+      httpHost =
+        attributes[ATTR_HTTP_HOST] ??
+        attributes[ATTR_SERVER_ADDRESS] ??
+        attributes[ATTR_CLIENT_ADDRESS];
+    }
+
+    let serviceType: AttributeValue | undefined = undefined;
+    let resourceARN: AttributeValue | undefined = undefined;
+
+    if (resource) {
+      serviceName = resource.attributes[ATTR_SERVICE_NAME] || '';
+      const cloudPlatform: AttributeValue | undefined =
+        resource.attributes[ATTR_CLOUD_PLATFORM];
+      if (typeof cloudPlatform === 'string') {
+        serviceType = CLOUD_PLATFORM_MAPPING[cloudPlatform];
+      }
+      resourceARN = this.getArn(resource, attributes);
+    }
+
+    // target may be in url
+    if (httpTarget === undefined && typeof httpUrl === 'string') {
+      const schemeEndIndex: number = httpUrl.indexOf('://');
+      // For network calls, URL usually has `scheme://host[:port][path][?query][#fragment]` format
+      // Per spec, url.full is always populated with scheme://
+      // If scheme is not present, assume it's bad instrumentation and ignore.
+      if (schemeEndIndex > -1) {
+        // urlparse("scheme://netloc/path;parameters?query#fragment")
+        httpTarget = new URL(httpUrl).pathname;
```

Comment:
Ensure that this will never receive a malformed `httpUrl` as it will throw if it does.